Fixes issue w/ BatchInserter in enterprise setting

tinwelint opened the pull request:
namely that some kernel extensions that was loaded required additional
dependencies, which the BatchInserter didn't have available already.
KernelExtensions must pull out all dependencies in `newInstance` method,
not later when initializing the actual extension lifecycle instance,
otherwise exception flow trips up and unsatisfied exception handler won't
be called correctly.

davidegrohmann commented on community/kernel/src/main/java/org/neo4j/kernel/impl/util/NullJobScheduler.java:
```diff
@@ -0,0 +1,87 @@
+/**
+ * Null {@link JobScheduler} which doesn't do anything with the jobs that gets scheduled.
+ */
+public class NullJobScheduler extends LifecycleAdapter implements JobScheduler
```
Why not using `OnDemandJobScheduler`? (it does nothing unless you call `runJob`)

tinwelint replied:
Alright

davidegrohmann commented on community/kernel/src/main/java/org/neo4j/unsafe/batchinsert/BatchInserterImpl.java:
```diff
@@ -296,14 +299,15 @@ public Label apply( long from )
-                                return neoStores;
-                            }
-                        } );
+        deps.satisfyDependencies( fileSystem, config, logService, new NullJobScheduler(), new Monitors(),
```
I am not sure I like giving a `NullJobScheduler` to satisfy dependencies here...

tinwelint replied:
Should we instead prevent that kernel extension from loading? There's no means of doing that currently.

davidegrohmann replied:
I was wondering about that. Not loading them at all, if possible, looks a much safer solution.

tinwelint replied:
Oh wait there's already a facility for doing that inside KernelExtensions, it's just bypassed somehow... I'll fix just that instead and remove the NulljobScheduler.

davidegrohmann commented on enterprise/neo4j-enterprise/src/test/java/soak/BatchInsertEnterpriseTest.java:
```diff
@@ -0,0 +1,105 @@
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package soak;
```
Why not matching the same package of the `BatchInserter`, i.e., `org.neo4j.unsafe.batchinsert`?

tinwelint replied:
it crossed my mind. sure, doesn't matter for me

tinwelint committed:
Fixes issue w/ BatchInserter in enterprise setting

namely that some kernel extensions that was loaded required additional
dependencies, which the BatchInserter didn't have available already.
KernelExtensions must pull out all dependencies in `newInstance` method,
not later when initializing the actual extension lifecycle instance,
otherwise exception flow trips up and unsatisfied exception handler won't
be called correctly.

tinwelint commented:
@davidegrohmann there's a much better fix now

davidegrohmann commented:
@tinwelint looks good, I'll merge on green

davidegrohmann commented:
@tinwelint test failures

tinwelint commented:
@davidegrohmann must be unrelated. Triggered another run.
